C#: Fix loading correct solution file for projects in subdirectories

samuelll3d opened the pull request:
Fixes: #98804

I was working on a multi-project solution and couldn't get my project to build and saw that there was an open pull request at #98807 and since it's been about 3 months now, I decided to continue the work on it a little myself. I've updated the FindSolutionFileWithAssemblyName method to make use of the [Microsoft.VisualStudio.SolutionPersistence](https://www.nuget.org/packages/Microsoft.VisualStudio.SolutionPersistence) package, which has support for both the `.sln` and the new `.slnx` format. This pull request does enable loading slnx formats, and I've been able to build a few test projects successfully using the format, though there may be other areas that need updating to properly add support for them. 

As with the previous pull request, this also enables building multi-project solutions. I've also gone through the changes requested to the previous pull request and tried to implement those here.

raulsntos reviewed (changes requested):
Thanks for contributing to the .NET module! The .NET team discussed this and other related issues/PRs, and we think this aligns with the direction we want to take.

We considered other proposals that suggested to add a new setting for the solution name (https://github.com/godotengine/godot/pull/77822) or a new setting for the full solution path that would replace the existing setting. But we think these options are confusing to users because the editor doesn't have a mechanism to deprecate settings, and we don't really like the idea of adding another setting for this.

We think the solution implemented by this PR is good enough, and we like how it handles errors when multiple matching solutions are found. There's only a few adjustments we'd like you to make.

Also, since you are a new contributor, make sure to read [CONTRIBUTING.md](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md) and the [contributing documentation](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html) if you haven't already.

You'll need to squash the commits before this PR can be merged. The [contributing documentation](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) contains information about squashing in case you need it.

Feel free to reach out in the [development chat](https://chat.godotengine.org) if you need help.

samuelll3d committed:
Find solution file by project assembly name

Instead of looking for a solution file with the same name as the project and
its assembly, this commit updates the logic to find all .sln and .slnx files in
the specified solution directory. If no matching solution is found, it will fall
back to the old behaviour.

This commit will also consider .. markings to go up one directory level allowing
for Godot projects as part of multi-project solutions.

Co-authored-by: Eric Johnson <eric.johnson@revention.com>

raulsntos commented:
I think it's fine, only a little annoying that we show the error for every C# file instead of only once. But it's the same as before this PR when you were missing the `.sln` file. This error is added here:

https://github.com/godotengine/godot/blob/e1b4101e3460dd9c6ba0b7f8d88e9751b8383f5b/modules/mono/editor/GodotTools/GodotTools/Export/ExportPlugin.cs#L132-L137

I don't think it's within the scope of this PR, so I would leave it as-is.

samuelll3d commented:
Alright, I'll just let that be then. I've also just tested this on Windows and updated the path separator handling to work properly. The error message will now also use Unix path separators to match the trace path separators.
